### PR TITLE
added blacklist to auto tool

### PR DESCRIPTION
**Pull request by TaxEvasiqn**
description

### CODE REVIEW

**Comment by TaxEvasiqn** [comment]
fixes https://github.com/MeteorDevelopment/meteor-client/issues/2963 or something i guess

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoTool.java [hunk 1]
```diff
@@ -173,12 +180,11 @@ public static double getScore(ItemStack itemStack, BlockState state, boolean sil
...
             score += 9000 + (item.getMaterial().getMiningLevel() * 1000);
 
-
         return score;
     }
 
-    public static boolean isTool(ItemStack itemStack) {
-        return itemStack.getItem() instanceof ToolItem || itemStack.getItem() instanceof ShearsItem;
+    public static boolean isTool(Item item) {
+        return item instanceof ToolItem || item instanceof ShearsItem;
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoTool.java:187]: Why did you change it to take in `Item`? From my perspective a useless change that only breaks compatibility with third party code.

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoTool.java [hunk 2]
```diff
@@ -123,6 +129,7 @@ private void onStartBreakingBlock(StartBreakingBlockEvent event) {
 
         for (int i = 0; i < 9; i++) {
             double score = getScore(mc.player.getInventory().getStack(i), blockState, silkTouchForEnderChest.get(), prefer.get(), itemStack -> !shouldStopUsing(itemStack));
+            if (blacklist.get().contains(mc.player.getInventory().getStack(i).getItem())) score = -1;
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoTool.java:132]: Can you move this before the `getScore()` call and return early so it doesn't need to run if it is already blacklisted. And while you are at it put the item inside a variable just so it is easier to read than twice repeating `mc.player.getInventory().getStack(i)`.

**Comment by MineGame159** [comment]
Closed in favor of cd739b56188b61f08cf2e3398b2457e4664b7b52.
